app/src/pages/inside/dashboardItemPage/widgetsGrid/widget/widget.jsx

```diff
@@ -1,52 +1,103 @@
-import React, { PureComponent } from 'react';
+import React, { Component, Fragment } from 'react';
 import classNames from 'classnames/bind';
 import PropTypes from 'prop-types';
+import { fetch } from 'common/utils';
+import { URLS } from 'common/urls';
+import { connect } from 'react-redux';
+import { activeProjectSelector } from 'controllers/user';
+import { showModalAction } from 'controllers/modal';
+import { SpinningPreloader } from 'components/preloaders/spinningPreloader';
+import { WidgetHeader } from './widgetHeader';
 import styles from './widget.scss';
 
 const cx = classNames.bind(styles);
 
-export class Widget extends PureComponent {
+@connect(
+  (state, ownProps) => ({
+    url: URLS.widget(activeProjectSelector(state), ownProps.widgetId),
```

Comment:
I suggest adding `widgetId` to propTypes to make this dependency clear

Reply:
fixed